src/ProposalTypesConfigurator.sol

```diff
@@ -71,37 +76,56 @@ contract ProposalTypesConfigurator is IProposalTypesConfigurator {
         return _proposalTypes[proposalTypeId];
     }
 
+    /**
+     * @notice Get the scope that is assigned to a given proposal type.
+     * @param proposalTypeId Id of the proposal type.
+     * @param scopeKey The function selector + contract address that is the key for a scope.
+     * @return Scope struct of the scope.
+     */
+    function assignedScopes(uint8 proposalTypeId, bytes24 scopeKey) external view override returns (Scope memory) {
+        return _assignedScopes[proposalTypeId][scopeKey];
+    }
+
+    /**
+     * @notice Returns a boolean if a scope exists.
+     * @param key A function selector and contract address that represent the type hash, i.e. 4byte(keccak256("foobar(uint,address)")) + bytes20(contractAddress).
+     * @return boolean returns true if the scope is defined.
+     */
+    function scopeExists(bytes24 key) external view override returns (bool) {
+        return _scopeExists[key];
+    }
+
     /**
      * @notice Sets the scope for a given proposal type.
      * @param proposalTypeId Id of the proposal type.
-     * @param txTypeHash A function selector that represent the type hash, i.e. keccak256("foobar(uint,address)").
+     * @param key A function selector and contract address that represent the type hash, i.e. 4byte(keccak256("foobar(uint,address)")) + bytes20(contractAddress).
      * @param encodedLimit An ABI encoded string containing the function selector and relevant parameter values.
      * @param parameters The list of byte represented values to be compared against the encoded limits.
      * @param comparators List of enumuerated values represent which comparison to use when enforcing limit checks on parameters.
      */
     function setScopeForProposalType(
         uint8 proposalTypeId,
-        bytes32 txTypeHash,
+        bytes24 key,
         bytes calldata encodedLimit,
         bytes[] memory parameters,
         Comparators[] memory comparators
     ) external override onlyAdmin {
```

Comment:
is it explicit in specs that we need to have `onlyAdmin` and not also allow timelock?

Reply:
Yup, got confirmation by @yitongzhang that only the admin will be adding scopes.